src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs

```diff
@@ -24,9 +24,12 @@ public StringInfo() : this(string.Empty)
 
         public StringInfo(string value)
         {
-            this.String = value;
-            // TODO-NULLABLE: compiler doesn't see this field being initialized through property
-            _str = _str!;
+            // TODO-NULLABLE: This used to be: `this.String = value;`
```

Comment:
@safern, @krwq. I worked around the issue by just manually inlining the initialization as I don't think disabling the self-assignment warning is the right way to workaround this.

CC. @jaredpar. @cston. Do we have a good story to allow bypassing this particular type of error when the user knows that the field was initialized to non-null?

Reply:
To work around the uninitialized non-nullable field warning:
disable the warning (`#pragma warning disable 8618`), or use `null!` at the declaration (`private string _str = null!;`). 

---
In reply to: [270507076](https://github.com/dotnet/coreclr/pull/23560#discussion_r270507076) [](ancestors = 270507076)

Reply:
Thanks! Updated to do `_str = null!`

Reply:
using `null!` looks like a good suggestion, @krwq could we do that in all your other changes? 😄